### PR TITLE
Add some more tests

### PR DESCRIPTION
**Pull request by ishmam-mahmud**
Add some more tests
- [x] add a README.
- [x] v1 release too

### CODE REVIEW

File: packages/ignore/README.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  pathsToIgnore: [],
+  pathsToSlice: ["allow_slicing"],
+  files: // list of files in the repo
+})
+
+// Output all the files that should be sliced from the repo.
+console.log(output.filesToSlice)
+```
+
+Note that you can call this library on either the sliced repo or the upstream repo, as the same gitslice config should work for both. You may need to do both repos for specific steps of the gitslice push/pull operations.
```
Comment (ethanppl) [inline packages/ignore/README.md:31]: What do you mean by calling this library on either the sliced or the upstream? Don't we always need to call on the upstream repo? And when do we have to call it on the sliced repo?

File: packages/ignore/index.test.ts [hunk 2]
```diff
@@ -127,6 +127,97 @@ describe("ignore mode", () => {
...
+      gitslice({
+        mode: "ignore",
+        pathsToIgnore: [],
+        pathsToSlice: ["/"],
+        files: ["a", "a/b", "c", "d/e"],
+      }),
+    ).toEqual<GitSliceOutput>({
+      filesToSlice: [],
+    });
+  });
```
Comment (ethanppl) [inline packages/ignore/index.test.ts:220]: Why is `/` not the same as `/*`?

**Review by ethanppl — COMMENTED** [review]
I have some questions

File: README.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
+# GitSlice
+
+This repo provides two packages
+
+- [@gitstart/gitslice-ignore](./packages/ignore/) - The reference implementation of the slicing algorithm that controls what should be sliced and what should not be.
```
Comment (richardguerre) [inline README.md:5]: I believe these relative links don't work in NPM, so we might need to use GitHub permalinks instead.

File: packages/ignore/README.md (hunk 1 above)
Comment (ethanppl) [reply packages/ignore/README.md:31]: Got it, thanks

**Review by ethanppl — APPROVED** [review]

**Pull request merged**